### PR TITLE
tls: support dual ECDSA/RSA certs via SDS

### PR DESCRIPTION
**Pull request by ggreenway**
This was previously only supported via static file-based certificate configuration.

Signed-off-by: Greg Greenway <ggreenway@apple.com>

Commit Message:
Additional Description:
Risk Level: Low
Testing: Added integration test; possibly need more unit tests
Docs Changes: Documented in protos
Release Notes: Added
Platform Specific Features: none
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -74,6 +74,7 @@ New Features
 * listener: added ability to change an existing listener's address.
 * metric service: added support for sending metric tags as labels. This can be enabled by setting the :ref:`emit_tags_as_labels <envoy_v3_api_field_config.metrics.v3.MetricsServiceConfig.emit_tags_as_labels>` field to true.
 * tcp: added support for :ref:`preconnecting <v1.18.0:envoy_v3_api_msg_config.cluster.v3.Cluster.PreconnectPolicy>`. Preconnecting is off by default, but recommended for clusters serving latency-sensitive traffic.
+* tls: allow dual ECDSA/RSA certs via SDS. This was previously only supported via static file-based certificate configuration.
```
Comment (htuch) [inline docs/root/version_history/current.rst:77]: That's not true; it was possible to do inline ECDS/RSA certs via TLS contexts included directly in `Listener` or `Cluster` transport socket configuration. Suggest rephrasing to reflect this.

**Review by htuch — CHANGES_REQUESTED** [review]
LGTM modulo nit on version history comment.

File: docs/root/version_history/current.rst (hunk 1 above)
Comment (ggreenway) [reply docs/root/version_history/current.rst:77]: Ah, I see. That's what I intended to say, but the phrasing came out misleading. Will fix.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Comment by ramaraochavali** [comment]
@ggreenway Sorry to tag you on old PR. This PR only enabled dual certificate in sds_secret_config but did not touch Validation Context. Is that assumption that CA will have single ca certificate for both RSA and ECDSA certs that it issues? Is that the normal case?

**Comment by ramaraochavali** [comment]
Thank you. That is what I realized too.

**Pull request merged**